openedx/core/djangoapps/util/maintenance_banner.py

```diff
@@ -0,0 +1,39 @@
+"""
+View decorator to add a maintenance banner configured in settings.
+"""
+from __future__ import absolute_import
+
+from django.conf import settings
+
+from openedx.core.djangoapps.util.user_messages import PageLevelMessages
+from openedx.core.djangoapps.util.waffle import DISPLAY_MAINTENANCE_WARNING, waffle
+
+def add_maintenance_banner(func):
+    """
+    View decorator to select where exactly the banner will appear
+
+    Add to function-based views like this:
+
+        from openedx.core.djangoapps.util.maintenance_banner import add_maintenance_banner
+
+        @add_maintenance_banner
+        def my_view(request):
+            ...
+
+    Add to class-based views using method_decorator:
+
+        from openedx.core.djangoapps.util.maintenance_banner import add_maintenance_banner
+        from django.utils.decorators import method_decorator
+
+        @method_decorator(add_maintenance_banner, name='dispatch')
+        class MyView(View):
+            ...
+    """
+    def _decorated(request, *args, **kwargs):
+        if waffle().is_enabled(DISPLAY_MAINTENANCE_WARNING):
+            if hasattr(settings, 'MAINTENANCE_BANNER_TEXT') and settings.MAINTENANCE_BANNER_TEXT:
```

Comment:
It's looking more like I should be prefixing this setting with EDXAPP to be more consistent with the names of other settings variables.

Reply:
I assume there is a companion PR somewhere with the MAINTENANCE_BANNER_TEXT?

Reply:
Yep, I'm putting that together now.  It will be two PRs, one for edx-internal, and another for edge-internal.